benchmarks/src/main/scala/scalaz/zio/IODeepAttemptBenchmark.scala

```diff
@@ -40,6 +41,68 @@ class IODeepAttemptBenchmark {
     Await.result(descend(0), Inf)
   }
 
+  @Benchmark
+  def completableFutureDeepAttempt(): BigInt = {
+    import java.util.concurrent.CompletableFuture
+
+    def descent(n: Int): CompletableFuture[BigInt] =
+      if (n == depth) {
+        val f = new CompletableFuture[BigInt]()
+        f.completeExceptionally(new Exception("Oh noes!"))
+        f
+      } else if (n == halfway) {
+        descent(n + 1).exceptionally(_ => 50)
+      } else {
+        descent(n + 1).thenApply(_ + n)
+      }
+
+    descent(0).get()
```

Comment:
This does not look lazy to me. Is there some way you can fix that?

Reply:
both CompletableFuture, and twitter.Future are eagerly evaluated
and are a reflection of scala Future benchmarks implementation

i don't really see a way to change this behaviour